architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+
+### Error Handling
+<!--
+Describe the error scenarios that may occur and the corresponding recovery/error handling and user experience.
+-->
+
+Radius would expect the `runtimes` property to have the correct format. We would return errors if the formatting is incorrect (this would be surfaced during the merge process) or when deploying the container group profile (this would be surfaced by ARM if the ACI SDK throws an error on creation). 
+
+## Test plan
+
+<!--
+Include the test plan to validate the features including the areas that
+need functional tests.
+
+Describe any functionality that will create new testing challenges:
+- New dependencies
+- External assets that tests need to access
+- Features that do I/O or change OS state and are thus hard to unit test
+-->
+
+TBD
+
+## Security
+
+<!--
+Describe any changes to the existing security model of Radius or security 
+challenges of the features. For each challenge describe the security threat 
+and its mitigation with this design. 
+
+Examples include:
+- Authentication 
+- Storing secrets and credentials
+- Using cryptography
+
+If this feature has no new challenges or changes to the security model
+then describe how the feature will use existing security features of Radius.
+-->
+
+The CCE policy is a base64 encoded string, however this is not considered a secret value. 
+
+## Compatibility (optional)
+
+<!--
+Describe potential compatibility issues with other components, such as
+incompatibility with older CLIs, and include any breaking changes to
+behaviors or APIs.
+-->
+
+This shouldn't cause any breaking changes to other components since it will be an added feature.
+
+## Monitoring and Logging
+
+<!--
+Include the list of instrumentation such as metric, log, and trace to 
+diagnose this new feature. It also describes how to troubleshoot this feature
+with the instrumentation. 
+-->
+
+## Development plan
+
+<!--
+Describe how you will deliver your features. This includes aligning work items
+to features, scenarios, or requirements, defining what deliverable will be
+checked in at each point in the product and estimating the cost of each work
+item. Don’t forget to include the Unit Test and functional test in your
+estimates.
+-->
+
+
+
+## Open Questions
+
+<!--
+Describe (Q&A format) the important unknowns or things you're not sure about. 
+Use the discussion to answer these with experts after people digest the 
+overall design.
+-->
+
+Is there an easier way to generate the CCE policy? The CLI extension is the only way to generate the CCE policy right now. There are no SDKs or APIs by design. 
+
+## Alternatives considered
+
+Option 1: 
+
+We could require the user to create a container group profile first and pass the resource to Radius. This would eliminate the need for Radius to install the `confcom` extension, generating ARM templates, and extracting the CCE policy. 
```

Comment:
I think this could also be a good option for users to use. They may go with this option or let Radius create the container group profile. i.e. I think we should offer both.